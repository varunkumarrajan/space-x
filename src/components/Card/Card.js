// React
import React from 'react';
// Prop types
import PropTypes from 'prop-types'; // ES6
// Styled component
import styled from 'styled-components';
// Component
import Image from 'components/Image/Image';
// Device
import device from 'config/BreakPoints';

// Styled card wrapper
const CardWrapper = styled.div`
  background-color: #ffffff;
  margin: 10px;
  padding: 20px;
  transition: 0.3s;

  @media screen and ${device.mobile} {
    width: calc(100% - 20px);
  }

  @media screen and ${device.tablet} {
    width: calc(50% - 20px);
  }

  @media screen and ${device.desktop} {
    width: calc(25% - 20px);
  }

  @media screen and ${device.desktopL} {
    width: calc(25% - 20px);
  }
`;

// Styled card content
const CardContent = styled.div`
  padding-top: 10px;
`;

// Styled card heading
const CardHeading = styled.p`
  color: #398fd8;
  font-size: 18px;
  font-weight: bold;
`;

// Styled card details
const CardDetails = styled.div`
  margin: 5px 0;
`;

// Styled card list
const CardList = styled.ul`
  padding-left: 30px;
`;

// Styled card bold subheading
const CardBoldSubHeading = styled.span`
  font-weight: bold;
`;

// Styled card sub heading
const CardSubHeading = styled.span`
  margin-left: 10px;
`;

/**
 * @description Card component have an image, heading and subheading & styled card component with styled component
 * @param {object} { content }
 */
const Card = ({ content }) => (
  <CardWrapper>
    <Image
      content={{
        alt: content.alt,
        src: content.src,
      }}
    />
    <CardContent>
      <CardHeading>{content.missionName}</CardHeading>
      <CardDetails>
        <CardBoldSubHeading>{content.labels.missionId}:</CardBoldSubHeading>
        <CardSubHeading>
          <CardList>
            {content.missionIds &&
              content.missionIds.length !== 0 &&
              content.missionIds.map((id, ind) => <li key={ind}>{id}</li>)}
          </CardList>
        </CardSubHeading>
      </CardDetails>
      <CardDetails>
        <CardBoldSubHeading>{content.labels.year}:</CardBoldSubHeading>
        <CardSubHeading>{content.year}</CardSubHeading>
      </CardDetails>
      <CardDetails>
        <CardBoldSubHeading>{content.labels.launch}:</CardBoldSubHeading>
        <CardSubHeading>{content.sLaunch}</CardSubHeading>
      </CardDetails>
      <CardDetails>
        <CardBoldSubHeading>{content.labels.land}:</CardBoldSubHeading>
        <CardSubHeading>{content.sLand}</CardSubHeading>
      </CardDetails>
    </CardContent>
  </CardWrapper>
);

// Default export
export default Card;

// Default props
Card.defaultProps = {
  content: {
    alt: '',
    src: '',
    labels: {},
    missionName: '',
    missionIds: [],
    year: '',
    sLaunch: '',
    sLand: '',
  },
};

// Prop types
Card.propTypes = {
  content: PropTypes.shape({
    alt: PropTypes.string.isRequired,
    src: PropTypes.string.isRequired,
    labels: PropTypes.any.isRequired,
    missionName: PropTypes.string.isRequired,
    missionIds: PropTypes.any.isRequired,
    year: PropTypes.string.isRequired,
    sLaunch: PropTypes.string.isRequired,
    sLand: PropTypes.string.isRequired,
  }),
};
