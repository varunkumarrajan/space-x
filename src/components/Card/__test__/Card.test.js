// React
import React from 'react';
// Testing library
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

import Card from './../Card';

// Mock data
const cardMock = {
  alt: 'testing',
  src: 'testing.png',
  labels: {},
  missionName: 'testing',
  missionIds: [],
  year: '2021',
  sLaunch: 'true',
  sLand: 'true',
}

afterEach(cleanup);

describe('Card', () => {
  it('Should Render Card With Props', () => {
    const { container } = render(<Card content={cardMock} />);
    expect(container.children).not.toHaveLength(0);
  });

  it('Card Snapshot', () => {
    const card = renderer.create(<Card content={cardMock} />).toJSON();
    expect(card).toMatchSnapshot();
  });
});
