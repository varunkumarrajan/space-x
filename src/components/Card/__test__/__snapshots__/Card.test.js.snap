// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Card Snapshot 1`] = `
<div
  className="sc-gsTCUz cFIiNV"
>
  <img
    alt="testing"
    className="sc-bdfBwQ eVkeNQ"
    src="testing.png"
    title="testing"
  />
  <div
    className="sc-dlfnbm dXuYyU"
  >
    <p
      className="sc-hKgILt hfOryn"
    >
      testing
    </p>
    <div
      className="sc-eCssSg hTrBts"
    >
      <span
        className="sc-gKsewC jqTwMs"
      >
        :
      </span>
      <span
        className="sc-iBPRYJ hpasmu"
      >
        <ul
          className="sc-jSgupP itglwW"
        />
      </span>
    </div>
    <div
      className="sc-eCssSg hTrBts"
    >
      <span
        className="sc-gKsewC jqTwMs"
      >
        :
      </span>
      <span
        className="sc-iBPRYJ hpasmu"
      >
        2021
      </span>
    </div>
    <div
      className="sc-eCssSg hTrBts"
    >
      <span
        className="sc-gKsewC jqTwMs"
      >
        :
      </span>
      <span
        className="sc-iBPRYJ hpasmu"
      >
        true
      </span>
    </div>
    <div
      className="sc-eCssSg hTrBts"
    >
      <span
        className="sc-gKsewC jqTwMs"
      >
        :
      </span>
      <span
        className="sc-iBPRYJ hpasmu"
      >
        true
      </span>
    </div>
  </div>
</div>
`;
